lambda_functions/ml_service/EnrichItem.py

```diff
@@ -280,6 +281,68 @@ def store_itementities(event, context, is_test=False, session=None):
                 raise
 
 
+def store_itemtags(event, context, is_test=False, session=None):
+    """stores tags of an item
+
+    Parameters
+    ----------
+    event: dict, required
+        item
+        Tags
+
+        Event doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+
+    context: object, required
+        Lambda Context runtime methods and attributes
+
+        Context doc: https://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+
+    Returns
+    ------
+    API Gateway Lambda Proxy Output Format: application/json
+
+        Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
+    """
+
+    if session is None:
+        session = get_db_session(is_test, session)
+
+    # Store all entities of the item
+    for str_tag in event['Tags']:
+        tag = Tag()
+        # search for tag in database
+        try:
+            tag = tag_handler.get_tag_by_content(
+                str_tag, is_test, session)
+        except Exception:
+            # store tag in database
+            tag.id = str(uuid4())
+            tag.tag = str_tag
+            try:
+                update_object(tag, is_test, session)
+            except Exception as e:
+                logger.error(
+                    "Could not store tag. Exception: %s", e, exc_info=True)
+                raise
```

Comment:
Flow of code is error prone in my opinion as the exception might have other reasons than the item not existing.

I'd rather catch the specific exception that occurs if a tag doesn't exist or let the `get_tag_by_content` function return null if a tag doesn't exist and then continue from there on.

Reply:
get_tag_by_content now returns None if the tag is not existing